### PR TITLE
add flag for specifying trace upload location

### PR DESCRIPTION
**Pull request by TyOverby**

### CODE REVIEW

**Comment by TyOverby** [comment]
@dotnet/roslyn-perf

**Comment by TyOverby** [comment]
@dotnet-bot test perf-correctness please

File: src/Test/Perf/Runner/Program.cs [hunk 1]
```diff
@@ -20,8 +20,28 @@ private static bool LaunchedWithArgument(string arg)
...
+            n = (n == -1) ? args.IndexOf("--{key}") : n;
+
+            if (n == -1)
+            {
+                return otherwise;
+            }
+
+            if (args.Count == n)
+            {
+                return otherwise;
```
Comment (KevinH-MS) [inline src/Test/Perf/Runner/Program.cs]: Not critical, but should this report an error of some sort?

**Comment by KevinH-MS** [comment]
LGTM, but it would be better if we reportage errors/usage message if arguments are malformed.

File: src/Test/Perf/Runner/Program.cs [hunk 2]
```diff
@@ -20,8 +20,28 @@ private static bool LaunchedWithArgument(string arg)
             return Environment.GetCommandLineArgs().Contains($"/{arg}") ||
                    Environment.GetCommandLineArgs().Contains($"--{arg}");
         }
+
+        private static string ValueForCommandLineKey(string key, string otherwise=null)
+        {
+            var args = Environment.GetCommandLineArgs().ToList();
+            var n = args.IndexOf($"/{key}");
+            n = (n == -1) ? args.IndexOf("--{key}") : n;
```
Comment (rchande) [inline src/Test/Perf/Runner/Program.cs]: Wow

**Comment by rchande** [comment]
:+1:

File: src/Test/Perf/Runner/Program.cs (hunk 2 above)
Comment (TyOverby) [inline src/Test/Perf/Runner/Program.cs]: I've removed this in favor of the Mono.Options arg parser

File: src/Test/Perf/Runner/Program.cs [hunk 3]
```diff
@@ -20,8 +20,28 @@ private static bool LaunchedWithArgument(string arg)
...
+
+            if (n == -1)
+            {
+                return otherwise;
+            }
+
+            if (args.Count == n)
```
Comment (TyOverby) [inline src/Test/Perf/Runner/Program.cs]: This code is removed now, but you were right, it should have been "args.Count - 1 == n"

**Pull request merged**